plugins/llvm/llvm_binary.hpp

```diff
@@ -50,17 +52,29 @@ int distance(content_iterator<T> begin, content_iterator<T> end) {
     return n;
 }
 
-std::vector<MachOObjectFile::LoadCommandInfo> load_commands(const MachOObjectFile* obj) {
-    std::size_t cmd_count = 0;
-    if (obj->is64Bit())
-        cmd_count = obj->getHeader64().ncmds;
-    else
-        cmd_count = obj->getHeader().ncmds;
+template <typename T>
+int distance(iterator_range<T> range)
+{
+    auto begin = range.begin();
+    auto end = range.end();
+    std::error_code ec;
+    int n = 0;
+    while (begin != end) {   
+        ++n;
+        begin.increment(ec);
+        if (ec)
+            llvm_binary_fail(ec);              
+    }
+    return n;  
+}
+
+std::vector<MachOObjectFile::LoadCommandInfo> load_commands(const MachOObjectFile* obj)
```

Comment:
I think that we don't need this function any more. It is better just call `obj->load_commands()` an iterate over a provided range.